crates/ruff_linter/src/rules/pylint/rules/iteration_over_set.rs

```diff
@@ -46,7 +46,10 @@ pub(crate) fn iteration_over_set(checker: &mut Checker, expr: &Expr) {
         return;
     };
 
-    if set.iter().any(Expr::is_starred_expr) {
+    // Only suggest a fix if all elements are literals.
+    // This is because we can't determine if the set is used to de-dupe
+    // the output values of a call, operation, etc.
+    if !set.iter().all(Expr::is_literal_expr) {
         return;
     }
```

Comment:
Technically, this also applies to literals. E.g. it's possible that someone wrote `set(1, 2, 2)` where fixing it to `[1, 2, 2]` would result in a semantic change. 

That's why I'm inclined that we should instead use `ComparableExpr` to test if no two set entries are identical. See e.g. 

https://github.com/astral-sh/ruff/blob/aa0db338d935aa40fafaa05feff7343c42491d8e/crates/ruff_linter/src/rules/pyflakes/rules/repeated_keys.rs#L141

Reply:
Rule `B033` catches duplicate set literals, so I've made it so this only emits a diagnostic if the set literals are unique.

The alternative is that we can still emit a diagnostic _without a fix_ if the set literals are not unique. Let me know what you think! 😄